test/javascripts/turbolinks_test.coffee

```diff
@@ -122,243 +122,175 @@ describe 'Turbolinks', ->
         assert(yourCallback.calledOnce, 'Callback was not called.')
         done()
 
-  describe 'head link tag tracking', ->
-    it 'dispatches page:after-link-inserted event when inserting a link on navigation', (done) ->
-      linkTagInserted = sinon.spy()
-      $(document).on 'page:after-link-inserted', linkTagInserted
+  describe 'head asset tracking', ->
+    it 'refreshes page when a data-turbolinks-track value matches but src changes', (done) ->
+      visit url: 'singleScriptInHeadTwo', ->
+        visit url: 'singleScriptInHeadWithDifferentSourceButSameName', ->
+          assert(Turbolinks.fullPageNavigate.called, 'Should perform a full page refresh.')
+          done()
 
-      visit url: 'noScriptsOrLinkInHead', ->
-        assertLinks([])
-        visit url: 'singleLinkInHead', ->
-          assertLinks(['foo.css'])
-          assert.equal(linkTagInserted.callCount, 1)
+    it 'does not refresh page when new data-turbolinks-track values encountered', (done) ->
+      visit url: 'singleScriptInHead', ->
+        visit url: 'twoScriptsInHead', ->
+          assert(Turbolinks.fullPageNavigate.notCalled, 'Should not perform a full page refresh.')
           done()
 
-    it 'dispatches page:after-link-removed event when removing a link on navigation', (done) ->
-      linkTagRemoved = sinon.spy()
-      $(document).on 'page:after-link-removed', linkTagRemoved
+    describe 'using data-turbolinks-track="true"', ->
+      startFromFixture = (route) ->
+        fixtureHTML = ROUTES[route][2]
+        $fixture = $(fixtureHTML)
+        document.body.innerHTML = $fixture.find('body').html()
+        document.head.innerHTML = $fixture.find('head').html()
 
-      visit url: 'singleLinkInHead', ->
-        visit url: 'noScriptsOrLinkInHead', ->
-          assert.equal(linkTagRemoved.callCount, 1)
+      it 'refreshes page when a new tracked node is present', (done) ->
+        visit url: 'singleScriptInHeadTrackTrue', ->
+          assert(Turbolinks.fullPageNavigate.called, 'Should perform a full page refresh.')
           done()
 
-    it 'inserts link with a new href into empty head on navigation', (done) ->
-      visit url: 'noScriptsOrLinkInHead', ->
-        assertLinks([])
-        visit url: 'singleLinkInHead', ->
-          assertLinks(['foo.css'])
+      it 'refreshes page when an extant tracked node is missing', (done) ->
+        startFromFixture('twoScriptsInHeadTrackTrue')
+        visit url: 'singleScriptInHeadTrackTrue', ->
+          assert(Turbolinks.fullPageNavigate.callCount, 2, 'Should perform two full page refreshes.')
           done()
 
-    it 'inserts link with a new href into existing head on navigation', (done) ->
-      visit url: 'singleLinkInHead', ->
-        assertLinks(['foo.css'])
-        visit url: 'twoLinksInHead', ->
-          assertLinks(['foo.css', 'bar.css'])
+      it 'does not refresh page when tracked node are the same', (done) ->
```

Comment:
`have matching sources` is probably clearest, since it's actually what the lib cares about in this case.